packages/flutter_locations/lib/src/config/flutter_locations_config.dart

```diff
@@ -0,0 +1,132 @@
+/// The type of the map
+enum MapType {
+  /// The normal map
+  normal,
+
+  /// The satellite map
+  satellite,
+
+  /// The hybrid map
+  hybrid,
+
+  /// The terrain map
+  terrain,
+
+  /// The openstreetmap map, t
+  openstreetmap,
+}
+
+/// The configuration for the flutter locations
+class FlutterLocationsConfig {
```

Comment:
this should be called PlatformMapConfiguration. Also, make sure that whenever these options are defined, that they are also used within the component. If we simply define config but never apply them to either, we create "dead" configuration.